wcomponents-core/src/main/java/com/github/bordertech/wcomponents/Margin.java

```diff
@@ -111,7 +136,9 @@ private Margin(final Size top, final Size right, final Size bottom, final Size l
 	/**
 	 * A margin equal on all sizes.
 	 * @param all the size of the margin to be used on all sides of the component.
+	 * @deprecated Use Margin.ZERO, Margin.SMALL, Margin.MEDIUM, Margin.LARGE or Margin.XL.
 	 */
+	@Deprecated
 	public Margin(final Size all) {
```

Comment:
I can understand why we might deprecate the constructor and make it private as this would force projects to use the predefined objects and it is stops projects creating their own "all" margins.
However, we are just providing the "all" objects as a convenience and it is already a public constructor, so my first thoughts are we dont really need to deprecate it. Also, if projects want to extend `Margin` for some reason they would need access to the constructor. If we do deprecate it, maybe it is better to say it will change to `protected` in the future. Also, in the unlikely event we add another `size` to the enum we would have to make sure we make the equivalent Margin object. 
My initial reaction is to not deprecate the constructor. Let me know what you think....

Reply:
Sure. I removed the deprecation from the constructor.

Reply:
I will add the tests.